lib/px/service/kinesis/base_request.rb

```diff
@@ -36,6 +36,7 @@ def initialize
       @last_throughput_exceeded = nil
 
       @buffer = []
+      @semaphore = Mutex.new
```

Comment:
Should we make it a class level variable? It doesn't protect the critical region if there are multiple client instances.